src/dash-table/derived/table/fragments.tsx

```diff
@@ -46,15 +47,23 @@ export default (
             cells) :
         null;
 
+    cells = R.addIndex<JSX.Element[]>(R.map)((row, i) => row.slice(
+        (fixedColumnCells && fixedColumnCells[i].length) || 0
+    ), cells);
```

Comment:
Fixed columns is impacted by merged headers, re-use the previous calculation to slice in the right place